src/main/java/com/mraof/minestuck/network/computer/BurnDiskPacket.java

```diff
@@ -50,24 +48,13 @@ private void tryBurnDisk(ComputerBlockEntity computer)
 	{
 		Item disk = this.isClientDisk ? MSItems.CLIENT_DISK.get() : MSItems.SERVER_DISK.get();
 		Level level = computer.getLevel();
-		BlockPos bePos = computer.getBlockPos();
 		if(level == null)
 			return;
 		
 		if(!computer.getProgramData(ProgramTypes.DISK_BURNER).map(DiskBurnerData::hasAllCode).orElse(false))
 			return;
 		
-		if(computer.tryTakeBlankDisk())
-		{
-			RandomSource random = level.getRandom();
-			
-			float rx = random.nextFloat() * 0.8F + 0.1F;
-			float ry = random.nextFloat() * 0.8F + 0.1F;
-			float rz = random.nextFloat() * 0.8F + 0.1F;
-			
-			ItemEntity entityItem = new ItemEntity(level, bePos.getX() + rx, bePos.getY() + ry, bePos.getZ() + rz, disk.getDefaultInstance());
-			entityItem.setDeltaMovement(random.nextGaussian() * 0.05F, random.nextGaussian() * 0.05F + 0.2F, random.nextGaussian() * 0.05F);
-			level.addFreshEntity(entityItem);
-		}
+		if(computer.canTakeDisk(MSItems.BLANK_DISK.get()))
+			computer.dropDisk(disk.getDefaultInstance());
```

Comment:
This should take a blank disk from the computer and then drop a program disk. Right now, it tries to take a program disk.

Reply:
Addressed!